interval/index.d.ts

```diff
@@ -0,0 +1,9 @@
+import { Event, Store } from 'effector';
+
+export function interval(config: {
+  timeout: number | Store<number>;
+  leading?: boolean;
+  trailing?: boolean;
+  start: Event<void>;
```

Comment:
I suggest forbidding `{leading: true, trailing: false }` options. Maybe something like
```
type Config = { leading: boolean } | { trailing: boolean }
```

?

Reply:
But, it also can be `{ leading: true, trailing: true }`

Reply:
We can change it to:

```tsx
type Config = {
  leading?: true;
  trailing?: true;
}
```

Reply:
Yeah, maybe something like it.

Reply:
But, if a user wants to create a variable with the condition to use `leading` or not, he will have type check issues.

```ts
const leading = someVar === 1 || process.env.DEBUG; // boolean

interval({
  timeout: 100,
  start: someEvent,
  leading,
})
```